pandas/core/arrays/base.py

```diff
@@ -610,3 +618,121 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionOpsBase(object):
```

Comment:
this should not contain a scalar based implemetation, if you really want to add that make it a separate MixIin (inherit from object)

Reply:
Note this is currently *not* a "base" class as what you have in mind, so it might be you just find the naming wrong?

Reply:
it IS a Base class for the arithemetic / comparison ops. This is the wrong place to put this logic. This needs to return NotImplementeError. as I have said multiple times.

Reply:
@jreback see my comment https://github.com/pandas-dev/pandas/pull/21261#discussion_r192418506 .  The class `ExtensionOpsBase` (and I'm open to new names) is not meant to be used directly as a parent class.  So why would it need to return NotImplementedError?  In fact, if someone just puts `ExtensionOpsBase` as a parent, it will do nothing beyond defining `create_method()`.